### ISSUE TITLE
Failed to recognize the category of a JSON file

### ISSUE THREAD

**Issue by SPGoding** [SPGoding/datapack-json#10]
# Description
When an advancement file is put under a directory like this: `data/<namespace>/advancements/recipes/`, it will be recognized as a recipe.

# Diagnostic

The current extension uses [following configuration](https://github.com/SPGoding/datapack-json/blob/ae7a45daa658462f4c78f9c4f669fc86208252f9/package.json#L37) to provide JSON validations: 
```json
"jsonValidation": [
    ...
    {
        "fileMatch": "data/*/advancements/*.json",
        "url": "./src/advancement.json"
    },
    ...
    {
        "fileMatch": "data/*/recipes/*.json",
        "url": "./src/recipe.json"
    },
    ...
]
```

The ideal solution is to use a glob pattern like `data/*/advancements/**/*.json` to match files. But unfortunately, the `fileMatch` doesn't support double asterisks (`**`) for recursion purposes, that said, `*` _indicates recursions_ already.

I have no idea about how to fix this problem, so I just open an issue and hope someone to help me out :/

**Comment by SPGoding** [comment]
This will be easily fixed once microsoft/vscode#74744 is resolved.

**Comment by SPGoding** [comment]
This is hard to fix by now. The whole JSON part of DHP will be combined in the language server in the future. See SPGoding/datapack-language-server#342.
